includes/functions.php

```diff
@@ -1478,12 +1478,22 @@ function get_attribution_domains() {
  * @return string The upload base URL.
  */
 function get_upload_baseurl() {
-	$upload_dir = \wp_get_upload_dir();
+	/**
+	 * Early filter to allow plugins to set the upload base URL.
+	 *
+	 * @param string|false $maybe_upload_dir The upload base URL or false if not set.
+	 */
+	$maybe_upload_dir = apply_filters( 'pre_activitypub_get_upload_baseurl', false );
```

Comment:
`pre_activitypub_get_upload_baseurl` or `activitypub_pre_get_upload_baseurl`???

Reply:
Core always uses a `pre_` prefix to indicate a bypass filter so I used that here.